core/src/main/java/org/apache/druid/collections/DefaultBlockingPool.java

```diff
@@ -86,6 +86,7 @@ public int getPoolSize()
       return wrapObject(timeoutMs > 0 ? pollObject(timeoutMs) : pollObject());
     }
     catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
```

Comment:
How about just propagate InterruptedException to callers? It makes more sense to me for all methods of BlockingPool.

Reply:
Do you mean using `throw Throwables.propagate(e);`?

Reply:
No, I mean making `InterruptedException` as a checked exception for all methods of BlockingPool.

Reply:
BTW, I'm fine by making another PR for this change. This PR is still worth without it.

Reply:
> BTW, I'm fine by making another PR for this change. This PR is still worth without it.

@leventov this PR is closed due to inactivity. However, it can be merged since https://github.com/apache/incubator-druid/pull/7121 has merged too?

Reply:
Yes. Please fix the conflicts.